flow-server/src/test/java/com/vaadin/flow/server/communication/IndexHtmlRequestHandlerTest.java

```
@@ -103,6 +108,34 @@ public void serveIndexHtml_requestWithRootPath_serveContentFromTemplate()
                 indexHtml.contains(".v-system-error"));
     }
 
+    @Test
+    public void serveNotFoundIndexHtml_requestWithRootPath_failsWithIOException() {
+        VaadinServletRequest vaadinServletRequest = createVaadinRequest("/");
+
+        VaadinService vaadinService = vaadinServletRequest.getService();
+        String indexHtmlPathInProductionMode = VAADIN_SERVLET_RESOURCES
+                + INDEX_HTML;
+
+        URL url = vaadinService.getClassLoader().getResource(indexHtmlPathInProductionMode);
+
+        try {
+            Objects.requireNonNull(url);
+            File indexHtmlFile = new File(url.getPath());
+            indexHtmlFile.delete();
+        } catch (Exception ignored) {}
```

Why are you ignore the exceptions? `Try-catch` should be excluded and those exceptions should be added to method signature, and if being thrown, fails the test.

---

Fixed